jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/WorkspaceConverterMojo.java

```diff
@@ -185,37 +191,51 @@ public void execute() throws MojoExecutionException, MojoFailureException {
-
-        Map<String, Object> lernaJson = new LinkedHashMap<>();
-        lernaJson.put("useWorkspaces", true);
-        lernaJson.put("version", "0.0.0");
-        Map<String, Object> lernaCommandMap = new LinkedHashMap<>();
-        Map<String, Object> lernaRunMap = new LinkedHashMap<>();
-        lernaRunMap.put("stream", true);
-        lernaCommandMap.put("run", lernaRunMap);
-        lernaJson.put("command", lernaCommandMap);
-
-        FileUtils.write(new File(convertedWorkspaceTarget + "/lerna.json"), objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(lernaJson).concat("\n"));
+      if (!new File(convertedWorkspaceTarget).exists()) {
+        new File(convertedWorkspaceTarget).mkdirs();
       }
+      RootPackageJson rootPackageJson = new RootPackageJson(objectMapper, convertedWorkspaceTarget);
+      List<String> workspaces = moduleMappings.entrySet().stream()
+              .map(entry -> {
+                if (ModuleType.IGNORE == entry.getValue().getModuleType()) {
+                  return null;
+                } else if (ModuleType.AGGREGATOR == entry.getValue().getModuleType() && !isAggregatorToConvert(entry.getValue().getData())) {
+                  return null;
+                } else {
+                  return getPackageFolderName(entry.getKey());
+                }
+              })
+              .filter(Objects::nonNull)
+              .collect(Collectors.toList());
+      rootPackageJson.writePackageJson(workspaces);
+
+      File lockFile = new File("lock");
+      boolean tryWriting = true;
+      while (tryWriting) {
+        try (FileOutputStream fileOutputStream = new FileOutputStream(lockFile)) {
+          fileOutputStream.getChannel().lock();
+          InputStream eslintrc = getClass().getResourceAsStream("/net/jangaroo/jooc/mvnplugin/.eslintrc.js");
+          if (eslintrc != null) {
+            Files.copy(eslintrc, Paths.get(convertedWorkspaceTarget + "/.eslintrc.js"), StandardCopyOption.REPLACE_EXISTING);
+          }
+          InputStream lernaJson = getClass().getResourceAsStream("/net/jangaroo/jooc/mvnplugin/lerna.json");
+          if (lernaJson != null) {
+            Files.copy(lernaJson, Paths.get(convertedWorkspaceTarget, "/lerna.json"), StandardCopyOption.REPLACE_EXISTING);
+          }
+          InputStream gitignoreFile = getClass().getResourceAsStream("/net/jangaroo/jooc/mvnplugin/.gitignore");
+          if (gitignoreFile != null) {
+            Files.copy(gitignoreFile, Paths.get(convertedWorkspaceTarget, ".gitignore"), StandardCopyOption.REPLACE_EXISTING);
+          }
+          InputStream npmrc = getClass().getResourceAsStream("/net/jangaroo/jooc/mvnplugin/.npmrc");
+          if (npmrc != null) {
+            Files.copy(npmrc, Paths.get(convertedWorkspaceTarget, ".npmrc"), StandardCopyOption.REPLACE_EXISTING);
+          }
+          tryWriting = false;
+        } catch (OverlappingFileLockException e) {
+          // lock is already acquired, trying again
+        }
```

Comment:
I guess Thread.sleep should not be used here to avoid high CPU load? @fwienber what is the best practise here?